test/meta.cpp

```diff
@@ -960,7 +1064,7 @@ TEST_F(Meta, MetaDataSetterGetterWithRefAsMemberFunctions) {
     ASSERT_TRUE(data);
     ASSERT_NE(data, meta::data{});
     ASSERT_EQ(data.parent(), meta::resolve("setter_getter"));
-    ASSERT_EQ(data.type(), meta::resolve<int>());
+    ASSERT_EQ(data.type(), meta::resolve<const int&>());
```

Comment:
Here is another example of what I said before. This change introduced a sort of (let me say) _explosion of types_ to take in consideration when most of the times I use this system with trivial data structures or simple functions. I don't really want to care of the fact that an argument is a `const int &` instead of an `int`. As long as the meta ctor returns me the `int` meta type for it I'm fine in 99% of the times. In this case I've to take in consideration the whole complexity of the type system of the C++, that was exactly what I tried to hide behind a simpler layer.

I agree that support for references is a must-have, but I'd like to find a way that doesn't turn `meta` in a tool with a type system as complex as the language it abstracts.

Reply:
As I said in the previous comment, #4 should address most of these issues. You can that that directly now. Just checkout #4 and use pointer types instead.

You will see that you don't actually have to care about `const`-ness (const is not supported there, but still the concept should still apply) or if a object is a pointer. You just give it as an argument and things just work.

Copying the example I gave there for reference:

```
// Note 1: Pointer types now have their members iterable
meta::func f = meta::resolve<T>()::func("test");
meta::func f2 = meta::resolve<T*>()::func("test"); // is also valid and is the same as `f`
assert(f == f2);

// Note 2: `clazz` removes `*`
assert(meta::resolve<T*>().clazz() == meta::resolve<T>());

// Note 3: both should work the same and does the correct thing, either with `handle` or `any`
T* object = ...;
f.invoke(object);
f.invoke(*object);
```

Also, importantly in the first note, both `f1` or `f2` are actually the same object (extracted from the `clazz` of type `T`). When using as an argument or as a `this` (first argument of invoke), you don't care if it is a pointer or not. It will just work:

```
// the "pseudo"-argument of invoke represents the type contained in an `any`.
f1.invoke(T*) // works
f2.invoke(T*) // works
f1.invoke(T) // works
f2.invoke(T) // works
```

Reply:
Also of note:

```
ASSERT_EQ(data.type().clazz(), meta::resolve<int>());
```

`clazz()` can be used to return the same value as would be returned previously. This is a breaking change still, but eases the pain of it.